CONTRIBUTING.md

```diff
@@ -52,31 +52,73 @@ It is unnecessary to manually edit (rebase) your commit history. If, however, yo
 
 ### Building and testing locally
 
-As described in [the README](https://github.com/scikit-hep/awkward-1.0#installation-for-developers), you can build locally using
+As described in [the README](readme.md), Awkward Array is shipped as two packages: `awkward` and `awkward-cpp`. The `awkward-cpp` package contains the compiled C++ components required for performance. If you do not need to frequently modify this code, then it can simply be installed using `pip`.
 
+Subsequent steps require the generation of code and datafiles (kernel specification, header-only includes). This can be done with the `prepare` nox session:
 ```bash
-python localbuild.py --pytest tests
+nox -s prepare
 ```
 
-The `--pytest tests` runs the integration tests from the `tests` directory (drop it to build only).
+<details>
+
+The `prepare` session accepts flags to specify exact generation targets, e.g.
+```bash
+nox -s prepare -- --tests --docs
+```
+This can reduce the time taken to perform the preparation step in the event that only the package-building step is needed.
+
+</details>
+
+#### Installing the `awkward-cpp` package
+
+The C++ components can be installed by building the `awkward-cpp` package:
+```bash
+python -m pip install ./awkward-cpp
```

Comment:
I think we always want to see the compilation output. Without `-vv`, a compiler or linker error would be hidden; pip would just say that it couldn't install and that isn't helpful. Anybody who is doing this step at least has the potential of having the compiler/linker fail, since the fully tested code-platform combinations are already wheels.

```suggestion
python -m pip install -vv ./awkward-cpp
```

Reply:
I'd recommend one `-v`, but without it, pip _does_ dump the output if the build fails, it just hides it if it passes, so it seems to just "freeze" for a bit. "build" doesn't hide the output by default (and is both prettier and more succinct).

IIRC, some number of v's causes it to print insanely long "findlinks" output that has nothing to do with the build. So pretty sure you always only want one `-v`. The backend does not get any options to increase output with more v's, only pip (which a user doesn't normally care about).

Reply:
If it dumps the output on failure, then that's good enough now.

"Now" because the compilation is quick enough: a minute or so is a reasonable length of time for a no-information pause; five minutes wasn't. I guess I used to be antsy in that period of time, wanting to see the compilation percentage increase. I don't remember why I thought one `v` wasn't good enough. I must have settled on something that works.

So I'm going to resolve this conversation—don't add the "`-v`"; no change requested.